utils/vector.hpp

```diff
@@ -0,0 +1,366 @@
+#pragma once
+
+namespace nstd
+{
+template<class T>
+class Vector
+{
+public:
+    using uint = unsigned int;
```

Comment:
is this necessary? I thought `uint` was standard since C++ version something-or-other

Reply:
``uint`` is not used by all compilers, rip